tensorboard/plugins/graph/metadata.py

```diff
@@ -24,3 +24,10 @@
 # graph Summaries.
 # See `graphs_plugin.py` for details.
 PLUGIN_NAME = "graphs"
+
+# In the context of the data provider interface, tag name given to a
+# graph that logically applies at the run level. No other types of
+# graphs are currently supported; when added, they'll have type-specific
+# name scopes (like `"__op_graph__/foo"`, `"__conceptual_graph__/bar"`)
+# to preclude collisions with this hard-coded string.
+RUN_GRAPH_NAME = "__run_graph__"
```

Comment:
Is this string is purely an internal constant, i.e. can we safely change it because the `if tag is None:` conditions are always in the same process as the `if tag == metadata.RUN_GRAPH_NAME:` conditions?  Or, are you expecting the uploader to provide this tag name, so that it will be persisted in backing storage (making changes here dangerous/impossible)?

Reply:
This _will become_ a persistent constant that needs to be stable, yes.
You can see a preview of how this will work at this (unstable) branch:
<https://github.com/wchargin/tensorboard/tree/wchargin-dataclass-compat>

As of this commit (which is just a bug fix, not a semantics change),
it’s purely in-process, and can be an arbitrary non-empty string.